.circleci/config.yml

```diff
@@ -119,8 +119,21 @@ commands:
       - run: 
           name: Deploying to k8s cluster for service binary-academy
           command: |
-            echo $CA_CRT | base64 --decode > ca.crt
-            kubectl --server=${KUBE_SERVER} --certificate-authority=ca.crt --token=$SERVICEACCOUNT_TOKEN set image deployment/academy-binary-com academy-binary-com=${DOCKHUB_ORGANISATION}/binary-static-academy:${CIRCLE_SHA1}
+            for SERVER_ID in {1..5}
+            do
+              KUBE_SERVER_REF="KUBE_SERVER_$SERVER_ID"
+              SERVICEACCOUNT_TOKEN_REF="SERVICEACCOUNT_TOKEN_$SERVER_ID"
+              CA_CRT_REF="CA_CRT_$SERVER_ID"
+              if [ ! -z "${!KUBE_SERVER_REF}" ]
```

Comment:
why using negation in the IF statement? I think it must be like this:
`if [ -z "${!KUBE_SERVER_REF}" ]`
also, we don't need indirect expansion too:

```
            for SERVER_ID in {1..5}
            do
              KUBE_SERVER_REF="KUBE_SERVER_$SERVER_ID"
              SERVICEACCOUNT_TOKEN_REF="SERVICEACCOUNT_TOKEN_$SERVER_ID"
              CA_CRT_REF="CA_CRT_$SERVER_ID"
              if [  -z "${KUBE_SERVER_REF}" ]
              then
                echo "Deploying to cluster $SERVER_ID"
                CA_CRT="${CA_CRT_REF}"
                KUBE_SERVER="${KUBE_SERVER_REF}"
                SERVICEACCOUNT_TOKEN="${SERVICEACCOUNT_TOKEN_REF}"
                echo $CA_CRT | base64 --decode > ca.crt
                kubectl --server=${KUBE_SERVER} --certificate-authority=ca.crt --token=$SERVICEACCOUNT_TOKEN set image deployment/bitcoin-wallet-binary-com bitcoin-wallet-binary-com=${DOCKHUB_ORGANISATION}/binary-bitcoin-wallet:${CIRCLE_SHA1}
              fi
            done
```

Reply:
`-z` checks for if the string is empty. But what I want is the opposite, I want to check if the variable for `KUBE_SERVER_[i]` is not empty and get the contents of it to proceed with the push to cluster.

Reply:
Yes, you're right about IF statement but what about indirect expansion?

Reply:
On first iteration `SERVER_ID=1`
`"${KUBE_SERVER_REF}"` Will expand to `"KUBE_SERVER_$SERVER_ID"` will expand to `"KUBE_SERVER_1` and then we check if the value of the variable $KUBE_SERVER_1 is not empty.